src/audio/mixin_mixout.c

```diff
@@ -1020,6 +1111,73 @@ static int mixout_get_attribute(struct comp_dev *dev, uint32_t type, void *value
 	return 0;
 }
 
+static int mixin_set_large_config(struct comp_dev *dev, uint32_t param_id, bool first_block,
+				  bool last_block, uint32_t data_offset_or_size, char *data)
+{
+	struct ipc4_mixer_mode_config *cfg;
+	struct mixin_data *mixin_data;
+	int i;
+	uint32_t sink_index;
+	uint16_t g;
+
+	if (param_id != IPC4_MIXER_MODE) {
+		comp_err(dev, "mixin_set_large_config() unsupported param_id: %u", param_id);
+		return -EINVAL;
+	}
+
+	if (!(first_block && last_block)) {
+		comp_err(dev, "mixin_set_large_config() data is expected to be sent as one chunk");
+		return -EINVAL;
+	}
+
+	/* for a single chunk data, data_offset_or_size is size */
+	if (data_offset_or_size < sizeof(struct ipc4_mixer_mode_config)) {
+		comp_err(dev, "mixin_set_large_config() too small data size: %u",
+			 data_offset_or_size);
+		return -EINVAL;
+	}
+
+	assert(data_offset_or_size <= SOF_IPC_MSG_MAX_SIZE);
+	dcache_invalidate_region((__sparse_force void __sparse_cache *)data, data_offset_or_size);
```

Comment:
is `data` a cached pointer? If it is, maybe the `.set_large_config` prototype should be changed to use `__sparse_cache`? In general - why is this needed? Is it shared between cores? Or accessed via cached and uncached aliases interchangeably?

Reply:
Large config payload comes from host via memory window. And so this invalidate cache is required. In fact, while developing in early version I did not have dcache_invalidate_region() call in my code and received garbage.

There are two possibilities, (1) to modify .set_large_config prototype to have __sparse_cache there, or (2) to call dcache_invalidate_region() internally before calling component's .set_large_config(). I think the second option might be nicer: I guess .set_large_config() is called from IPC task which always runs on core0.
But I'd suggest that this fix should be done as a separate PR and this PR could be merged as is.